### PR TITLE
Fix spelling and grammatical errors

### PR DESCRIPTION
**Pull request by nwithan8**
# Description

- Correct some functionality-effecting spelling errors (misspelled JSON elements)
- Corrected some non-functionality-effecting spelling and grammatical errors (i.e. docstrings)

# Testing

All cassettes still work, JSON should parse correctly now.

# Pull Request Type

Please select the option(s) that are relevant to this PR.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement (fixing a typo, updating readme, renaming a variable name, etc)

### CODE REVIEW

File: rate.go [hunk 1]
```diff
@@ -25,7 +25,7 @@ type Rate struct {
 	DeliveryDays           int        `json:"delivery_days,omitempty"`
 	DeliveryDate           *time.Time `json:"delivery_date,omitempty"`
 	DeliveryDateGuaranteed bool       `json:"delivery_date_guaranteed,omitempty"`
-	EstDeliveryDays        int        `json:"est_delivery_dats,omitempty"`
+	EstDeliveryDays        int        `json:"est_delivery_dates,omitempty"`
```
Comment (Justintime50) [inline rate.go:28]: days, not dates

File: rate.go [hunk 2]
```diff
@@ -48,7 +48,7 @@ type SmartRate struct {
 	DeliveryDays           int            `json:"delivery_days,omitempty"`
 	DeliveryDate           *time.Time     `json:"delivery_date,omitempty"`
 	DeliveryDateGuaranteed bool           `json:"delivery_date_guaranteed,omitempty"`
-	EstDeliveryDays        int            `json:"est_delivery_dats,omitempty"`
+	EstDeliveryDays        int            `json:"est_delivery_dates,omitempty"`
```
Comment (Justintime50) [inline rate.go:51]: days, not dates

File: scan_form.go [hunk 3]
```diff
@@ -36,6 +36,7 @@ type ListScanFormsResult struct {
 	HasMore bool `json:"has_more,omitempty"`
 }
 
+// TODO: Fix function name
```
Comment (Justintime50) [inline scan_form.go:39]: In golang, lowercased function names are not public meaning you should be able to correct this without affecting users.

File: tests/report_test.go [hunk 4]
```diff
@@ -40,8 +40,8 @@ func (c *ClientTests) TestReportCustomColumnsCreate() {
 
 	// verify parameters by checking VCR cassette for correct URL
 	// Some reports take a long time to generate, so we won't be able to consistently pull the report
-	// There's unfortunately no way to check if the columns were included in the final report without parsing the CSV
-	// so we assume, if we haven't gotten an error by this point, we've made the API calls correctly
```
Comment (Justintime50) [inline tests/report_test.go:44]: Isn't this a continuation of the sentence? Uppercasing doesn't seem like the right change

**Review by Justintime50 — DISMISSED** [review]

**Review by Justintime50 — APPROVED** [review]
Thanks for this

**Review by ldewald — APPROVED** [review]

**Pull request merged**